### ISSUE TITLE
[Error] WP Offload Media Lite and AMP

### ISSUE THREAD

**Issue by tongdaitructuyen** [deliciousbrains/wp-amazon-s3-and-cloudfront#564]
- Wordpress 5.5
- WP Offload Media Lite - Version 2.4.2
- AMP - Version 2.0.0

It show error:

```
Fatal error: Uncaught Error: Class ‘DeliciousBrains\WP_Offload_Media\Aws3\Symfony\Polyfill\Intl\Idn\Idn’ not found in /home/public_html/wp-content/plugins/amazon-s3-and-cloudfront/vendor/Aws3/Symfony/Polyfill/IntlIdn/bootstrap.php:133 Stack trace: #0 /home/public_html/wp-content/plugins/amp/includes/class-amp-http.php(240): idn_to_utf8(‘domain.com’, 0, 1) #1 /home/public_html/wp-content/plugins/amp/includes/class-amp-http.php(205): AMP_HTTP::get_amp_cache_hosts() #2 /home/public_html/wp-includes/class-wp-hook.php(289): AMP_HTTP::filter_allowed_redirect_hosts(Array) #3 /home/public_html/wp-includes/plugin.php(206): WP_Hook->apply_filters(Array, Array) #4 /home/public_html/wp-includes/pluggable.php(1480): apply_filters(‘allowed_redirec…’, Array, ‘domain.com’) #5 /home/public_html/wp-includes/functions.php(1905): wp_validate_redirect(‘https://domain…’, false) #6 /home/public_html/wp-includes/pluggable.php(1139): wp_get_referer() #7 /home/public_html/wp-conte in /home/public_html/wp-content/plugins/amazon-s3-and-cloudfront/vendor/Aws3/Symfony/Polyfill/IntlIdn/bootstrap.php on line 133
There has been a critical error on your website. Please check your site admin email inbox for instructions.
```

**Comment by ianmjones** [comment]
This seems to be happening on sites that do not include the `idn_to_utf8` built in PHP function for some reason, AMP tries to use it, but the AWS PHP SDK embedded in WP Offload Media has already created a polyfill for it, but alas the PHP Scoper process we use to protect our embedded libraries from clashing with others has not understood the aliasing of the `use` when altering the `bootstrap.php` file.

All that to say, we're looking into it, but best bet until we can find a fix is to either ensure the `idn_to_utf8` PHP function is included on the server, switch back to WP Offload Media (Lite) 2.4.1, or temporarily deactivate the AMP plugin.

**Comment by ianmjones** [comment]
Fixed in 2.4.3.
